### PR TITLE
Add personal YMDK Split64 keymap

### PR DESCRIPTION
**Pull request by minkezhang**
## Description

Add personal YMDK Split64 keymap for [minkezhang](https://github.com/minkezhang)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Please review the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and verify that you have followed the checklist for the type of PR this is.


Namely, c and h files need licensing headers, and files should be all lower case.

**Comment by minkezhang** [comment]
Done. PR Checklist mentions "GPL-compatible" licenses, but is not explicit on the version. I assume since this implies there is GPLv2 code in the repo, we cannot use Apache2?

**Comment by minkezhang** [comment]
@drashna Gentle ping, anything else here that I should be on the lookout for?

File: keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  KC_BSPC, KC_EQUAL,  KC_MINUS, KC_0,    KC_9, KC_8,    KC_7,          _______, _______, _______, _______, _______, _______, _______,
+  KC_BSLS, KC_P,      KC_O,     KC_I,    KC_U, KC_Y,          _______, _______, _______, _______, _______, _______, _______, _______,
+  KC_ENT,  KC_SCOLON, KC_L,     KC_K,    KC_J, KC_H,          _______, _______, _______, _______, _______, _______,          _______,
+  _______, KC_SLSH,   KC_DOT,   KC_COMM, KC_M, KC_N,          _______, _______, _______, _______, _______, _______, _______, _______,
+  _______, _______,   _______,  _______,       _______,       _______,          _______, KC_NO,            _______, _______, _______
+  ),
+};
+
+#ifdef RGBLIGHT_ENABLE
+#ifdef RGBLIGHT_LAYERS
```
Comment (drashna) [inline keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c:116]:
```suggestion
#if defined(RGBLIGHT_ENABLE) && defined(RGBLIGHT_LAYERS)
```

File: keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+void keyboard_post_init_user(void) {
+    rgblight_layers = user_rgb_layers;
+}
+#endif  // RGBLIGHT_LAYERS
+#endif  // RGBLIGHT_ENABLE
+
+layer_state_t default_layer_state_set_user(layer_state_t state) {
+
+#ifdef RGBLIGHT_ENABLE
+#ifdef RGBLIGHT_LAYERS
```
Comment (drashna) [inline keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c:138]:
```suggestion
layer_state_t default_layer_state_set_user(layer_state_t state) {
```

File: keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    rgblight_set_layer_state(0, layer_state_cmp(state, BASE));
+#endif  // RGBLIGHT_LAYERS
+#endif  // RGBLIGHT_ENABLE
```
Comment (drashna) [inline keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c:141]:
```suggestion
```

File: keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+    return state;
+}
+
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (drashna) [inline keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c:146]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+/* Add some indication for current function layer.
+ *
+ * N.B.: rgblight_set_layer_state calls rgblight_mode_noeeprom.
+ */
+#ifdef RGBLIGHT_ENABLE
+#ifdef RGBLIGHT_LAYERS
```
Comment (drashna) [inline keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c:153]:
```suggestion
```

File: keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    rgblight_set_layer_state(1, layer_state_cmp(state, _FN));
+    rgblight_set_layer_state(2, layer_state_cmp(state, _NUM));
+    rgblight_set_layer_state(3, layer_state_cmp(state, _MIRROR));
+#endif  // RGBLIGHT_LAYERS
+#endif  // RGBLIGHT_ENABLE
```
Comment (drashna) [inline keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c:158]:
```suggestion
```

File: keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+    return state;
+}
```
Comment (drashna) [inline keyboards/ymdk/sp64/keymaps/minkezhang/keymap.c:161]:
```suggestion
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**